hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operand/AbstractOperandChecker.java

```diff
@@ -35,7 +35,7 @@ protected AbstractOperandChecker() {
     public boolean check(HazelcastCallBinding callBinding, boolean throwOnFailure, int operandIndex) {
         HazelcastSqlValidator validator = callBinding.getValidator();
 
-        SqlNode operand = callBinding.operand(operandIndex);
+        SqlNode operand = callBinding.getCall().getOperandList().get(operandIndex);
```

Comment:
What is the motivation for this change? I presume there is something wrong with `operand(operandIndex)`?

Reply:
`operand()` returns _permuted_ node - because `OperandChecker`s operate on indices and positional arguments it would mess things up in case of argument assignments are present.